src/effect/delay.rs

```diff
@@ -21,9 +20,11 @@ impl Delay {
         let mut audio_units = vec![];
         let mut message_senders = vec![];
 
-        for n in 1..=config.num {
+        for n in 0..config.num {
+            let delay = Self::delay_for_index(config.delay_ms, n);
+            let max_total_delay = config.max_delay_ms * config.num;
             let (delay_unit, messages) =
-                audio_unit::Delay::new(stream_config, config.delay_ms * n)?;
+                audio_unit::Delay::new(stream_config, delay, max_total_delay)?;
```

Comment:
I think you need to validate somewhere that the initially configured values are reasonable (I guess that `delay_ms * num <= max_delay_ms`, since otherwise the buffer won't be big enough)?

And per sort of the spirit of this PR, it seems like that "somewhere" could now be in this delay effect initialization (and keep the audio units dumb)?